packages/angular_devkit/core/src/json/schema/interface.ts

```diff
@@ -6,9 +6,11 @@
  * found in the LICENSE file at https://angular.io/license
  */
 import { Observable } from 'rxjs/Observable';
+import { JsonValue } from '..';
 
 
 export interface SchemaValidatorResult {
+  data: JsonValue;
```

Comment:
Does having the data be returned here means the original values are no longer altered? That would be useful overall. It's kinda unintuitive that the original is mutated now.

Reply:
You get a visit for the initial value, so if you return a different value then we cannot mutate the original data.

Making a copy of the data is impossible at that point, because some of it might not be JSON values but objects and whatnot.